oprf/client.go

```diff
@@ -0,0 +1,156 @@
+package oprf
+
+import (
+	"crypto/rand"
+	"errors"
+)
+
+// Client is a representation of a Client during protocol execution.
+type Client struct{ suite }
+
+// ClientRequest is a structure to encapsulate the output of a Request call.
+type ClientRequest struct {
+	inputs          [][]byte
+	blinds          []Blind
+	BlindedElements []Blinded
+}
+
+// NewClient creates a new instantiation of a Client.
+func NewClient(id SuiteID, m Mode) (*Client, error) {
```

Comment:
Should we pass in the public key here, in verifiable mode? (And perhaps default to base mode if the public key is omitted?)

```suggestion
func NewClient(id SuiteID) (*Client, error) {
   // base mode
}

func NewVerifiableClient(id SuiteID, pks *PublicKey) (*Client, error) {
   // verifiable mode
}
```